### PR TITLE
[Deno] Setup API type definition configuration

### PR DESCRIPTION
**Pull request by vyktoremario**
## Type of change

- [ ] Documentation change
- [ ] Bug fix
- [x] feature

## Summary of change

closes: #566 

## Checklist

- [x] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [x] This starter kit has been approved by the maintainers
- [x] Changes for this new starter kit are being pushed to an integration branch instead of main
- [x] All dependencies are version locked
- [ ] This fix resolves #
- [ ] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/deno-oak-denodb/tools/generateTypeDefinition.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { CodegenConfig, generate } from 'npm:@graphql-codegen/cli';
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/tools/generateTypeDefinition.ts:1]: Shouldn't this file be named `generate_type_definition.ts` according to the styleguide?

File: starters/deno-oak-denodb/src/graphql/interfaces/codegen.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  __typename?: 'Technology';
+  createdAt: Maybe<Scalars['String']>;
+  description: Maybe<Scalars['String']>;
+  displayName: Maybe<Scalars['String']>;
+  id: Maybe<Scalars['String']>;
+  updatedAt: Maybe<Scalars['String']>;
+  url: Maybe<Scalars['String']>;
+};
+
+export type TechnologyInput = {
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/graphql/interfaces/codegen.ts:65]: The types are generated here, but they don't seem to be used anywhere. Shouldn't they be used in other modules?

**Review by dariodjuric — CHANGES_REQUESTED** [review]
Left a couple comments.

File: starters/deno-oak-denodb/tools/generateTypeDefinition.ts (hunk 1 above)
Comment (vyktoremario) [reply starters/deno-oak-denodb/tools/generateTypeDefinition.ts:1]: I keep spelling that wrong😑

**Pull request merged**